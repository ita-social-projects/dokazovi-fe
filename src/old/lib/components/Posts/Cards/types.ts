import { IPost } from '../../../types';

export interface IPostPreviewCardProps {
  post: IPost;
  shouldNotUseLink?: boolean;
}
