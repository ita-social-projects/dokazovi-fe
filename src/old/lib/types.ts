export enum ExpertStatus {
  NEW = 'NEW',
  ACTIVE = 'ACTIVE',
  DELETED = 'DELETED',
}
export enum PostStatus {
  DRAFT = 'DRAFT',
  MODERATION_FIRST_SIGN = 'MODERATION_FIRST_SIGN',
  MODERATION_SECOND_SIGN = 'MODERATION_SECOND_SIGN',
  PUBLISHED = 'PUBLISHED',
  ARCHIVED = 'ARCHIVED',
}

export enum OriginTypeEnum {
  EXPERT_OPINION = 1,
  TRANSLATION = 2,
  MEDIA = 3,
}

export type OriginType =
  | OriginTypeEnum.EXPERT_OPINION
  | OriginTypeEnum.MEDIA
  | OriginTypeEnum.TRANSLATION;

export enum PostTypeEnum {
  ARTICLE = 1,
  DOPYS = 2,
  VIDEO = 3,
}

export enum LoadMoreButtonTextType {
  EXPERT,
  POST,
}

export enum PostsPreviewCardStylesEnum {
  VIDEO = 'VIDEO',
  MEDIA = 'MEDIA',
  TRANSLATION = 'TRANSLATION',
  EXPERT_OPINION = 'EXPERT_OPINION',
  IMPORTANT = 'IMPORTANT',
}

export type PostsPreviewCardStylesType =
  | PostsPreviewCardStylesEnum.VIDEO
  | PostsPreviewCardStylesEnum.MEDIA
  | PostsPreviewCardStylesEnum.TRANSLATION
  | PostsPreviewCardStylesEnum.EXPERT_OPINION
  | PostsPreviewCardStylesEnum.IMPORTANT;

export enum LoadingStatusEnum {
  idle = 'idle',
  pending = 'pending',
  succeeded = 'succeeded',
  failed = 'failed',
}

export type LoadingStatusType =
  | LoadingStatusEnum.idle
  | LoadingStatusEnum.pending
  | LoadingStatusEnum.succeeded
  | LoadingStatusEnum.failed;

export enum LocalStorageKeys {
  ACCESS_TOKEN = 'ACCESS_TOKEN',
}

export enum FilterTypeEnum {
  POST_TYPES,
  DIRECTIONS,
  REGIONS,
  TAGS,
  ORIGINS,
}

export enum ChipFilterEnum {
  REGION = 'REGION',
  DIRECTION = 'DIRECTION',
  ORIGIN = 'ORIGIN',
  POST_TYPE = 'POST_TYPE',
}

export type ChipFilterType =
  | ChipFilterEnum.REGION
  | ChipFilterEnum.DIRECTION
  | ChipFilterEnum.ORIGIN
  | ChipFilterEnum.POST_TYPE;

export enum QueryTypeEnum {
  POST_TYPES = 'types',
  DIRECTIONS = 'directions',
  REGIONS = 'regions',
  TAGS = 'tags',
  PAGE = 'page',
  ORIGINS = 'origins',
}

export interface IPost {
  id: number;
  title: string;
  content: string;
  author: {
    avatar?: string;
    firstName: string;
    id: number;
    lastName: string;
    mainInstitution: {
      city: {
        id: number;
        name: string;
      };
      id: number;
      name: string;
    };
  };
  directions: IDirection[];
  origin: IOrigin[];
  tags?: IPostTag[];
  type: IPostType;
  createdAt: string;
  modifiedAt?: string;
  origins?: IOrigin[];
  preview: string;
  previewImageUrl?: string;
  videoUrl?: string;
}

export interface IExpert {
  id: number;
  firstName: string;
  lastName: string;
  avatar?: string;
  qualification?: string;
  phone?: string;
  email?: string;
  bio?: string;
  mainInstitution?: IInstitution;
  mainDirection?: IDirection;
  directions?: IDirection[];
  lastAddedPost?: {
    id: number;
    title: string;
  };
}

export interface IPostTag {
  id: number;
  tag: string;
}

export interface IPostType {
  id: number;
  name: string;
}

export interface IDirection {
  id: number;
  color?: string;
  name: string;
  label?: string;
  hasPosts?: boolean;
}

export interface IFilter {
  id: number;
  color?: string;
  name: string;
  label?: string;
}

export interface IOrigin {
  id: number;
  name: string;
  parameter: null | string | number;
}

export interface IRegion {
  id: number;
  name: string;
  usersPresent?: boolean;
}

export interface IInstitution {
  city: {
    id: number;
    name: string;
  };
  id: number;
  name: string;
}

export interface ICheckboxes {
  [key: string]: boolean;
}

export interface IAuthInputs {
  email: string;
  password: string;
}
