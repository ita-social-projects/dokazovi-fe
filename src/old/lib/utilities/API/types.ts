import { IDirection, IOrigin, IPost } from '../../types';

export type CreatePostRequestType = {
  title: string;
  authorsName?: string;
  authorsDetails?: string;
  content: string;
  directions: IDirection[];
  origins?: IOrigin[];
  preview: string;
  type: {
    id: number;
  };
  previewImageUrl?: string;
  videoUrl?: string;
  authorId?: number | null;
};

export type CreateTextPostRequestType = CreatePostRequestType;

export type CreateVideoPostRequestType = CreatePostRequestType & {
  videoUrl: string;
};

export type CreatePostRequestUnionType =
  | CreateTextPostRequestType
  | CreateVideoPostRequestType;

type UpdatePostRequestType = CreatePostRequestType & {
  id: number;
};

export type UpdateTextPostRequestType = UpdatePostRequestType;

export type UpdateVideoPostRequestType = UpdatePostRequestType & {
  videoUrl: string;
};

export type UpdatePostRequestUnionType =
  | UpdateTextPostRequestType
  | UpdateVideoPostRequestType;

export type PostResponseType = {
  author: {
    avatar: string;
    firstName: string;
    id: number;
    lastName: string;
    mainInstitution: {
      city: {
        id: number;
        name: string;
      };
      id: number;
      name: string;
    };
  };
  content: string;
  preview: string;
  postType: PostTypeResponseType;
  createdAt: string;
  directions: DirectionResponseType[];
  origins: IOrigin[];
  id: number;
  title: string;
  type: {
    id: number;
    name: string;
  };
  videoUrl?: string;
};

export type ExpertResponseType = {
  avatar: string;
  firstName: string;
  id: number;
  lastAddedPost: {
    id: number;
    title: string;
  };
  lastName: string;
  mainDirection: {
    id: number;
    name: string;
  };
  mainInstitution: {
    city: {
      id: number;
      name: string;
    };
    id: number;
    name: string;
  };
  qualification: string;
};

export type CreateTagRequestType = {
  tag: string;
};

export type TagResponseType = {
  id: number;
  tag: string;
};

export type VersionResponseType = {
  version: string;
};

export type DirectionResponseType = {
  id: number;
  name: string;
  label: string;
  color: string;
};

export type OriginResponseType = {
  id: number;
  name: string;
  parameter: null;
};

export type RegionResponseType = {
  id: number;
  name: string;
};

export type PostTypeResponseType = {
  id: number;
  name: string;
};

export type LoginResponseType = {
  accessToken: string;
  tokenType: string;
};

type GetResponseType<T> = {
  content: T[];
  empty: boolean;
  first: boolean;
  last: boolean;
  number: number;
  numberOfElements: number;
  size: number;
  sort: {
    empty: boolean;
    sorted: boolean;
    unsorted: boolean;
  };
  totalElements: number;
  totalPages: number;
};

export type ExpertsResponseType = GetResponseType<ExpertResponseType>;

export type PostsResponseType = GetResponseType<PostResponseType>;

export type NewestPostsResponseType = GetResponseType<NewestPostResponseType>;

export enum NewestTypeEnum {
  MEDIA,
  TRANSLATION,
  EXPERT_OPINION,
  VIDEO,
}

export type NewestPostResponseType = {
  fieldName: string;
  postDTOS: IPost[];
};

export type GetTagsConfigType = {
  params: {
    value: string;
    limit?: number;
  };
};

type GetConfigType = {
  params: RequestParamsType;
};

export type GetExpertsConfigType = GetConfigType & {
  params: {
    directions?: number[];
    regions?: number[];
  };
};

export type GetPostsConfigType = GetConfigType;

export type RequestParamsType = {
  page?: number;
  size?: number;
  sort?: string[];
  directions?: number[];
  type?: number[];
  types?: number[];
  expert?: number;
  regions?: number[];
  tag?: number[];
  origins?: number[];
  userName?: string;
};
