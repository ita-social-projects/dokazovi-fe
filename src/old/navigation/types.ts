export interface IRouteConfig {
  component: React.ComponentType;
  path: string | string[];
  key: string;
  exact?: boolean;
  private?: boolean;
  useRender?: boolean;
}
