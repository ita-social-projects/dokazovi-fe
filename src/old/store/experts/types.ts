import { LoadingStatusEnum } from '../../lib/types';
import { IMaterials } from '../materials/types';

export interface IExpertPayload {
  expertIds: number[];
  meta: IExpertMeta;
}

export interface IExpertMeta {
  totalPages: number;
  pageNumber: number;
  loading: LoadingStatusEnum;
  error: null | string;
  isLastPage: boolean;
  totalElements: number;
}

export interface IExpertsState {
  experts: IExpertPayload;
  materials: IMaterials;
}

export interface IFetchExpertsOptions {
  page: number;
  regions: number[];
  directions: number[];
  appendExperts: boolean;
}
