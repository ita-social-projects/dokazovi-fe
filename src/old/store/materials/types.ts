import { QueryTypeEnum, LoadingStatusEnum, IPost } from '../../lib/types';

export interface IMaterials {
  postIds: number[];
  meta: IMaterialsMeta;
} // Old type, used in expertsSlice, it will be removed;

interface IData {
  postIds: number[];
  meta: IMaterialsMeta;
  posts: {
    [id: string]: IPost;
  };
}

export interface IMaterialsState {
  data: IData;
  loading: LoadingStatusEnum;
  error: null | string;
  filters?: {
    [key in QueryTypeEnum]?: number[];
  };
}

export interface IFetchMaterialsOptions {
  filters: {
    page: number;
    postTypes: number[];
    directions: number[];
  };
  page: number;
  appendPosts: boolean;
}

export interface IMaterialsMeta {
  loading: LoadingStatusEnum;
  error: null | string;
  isLastPage: boolean;
  pageNumber: number;
  totalPages: number;
  totalElements: number;
} // Properties loading & error are used in expertsSlice, they will be removed from this interface;
