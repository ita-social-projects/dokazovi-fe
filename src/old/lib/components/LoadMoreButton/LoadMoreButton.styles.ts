import { makeStyles, createStyles } from '@material-ui/core';

export const useStyles = makeStyles(
  (theme) =>
    createStyles({
      root: {
        width: '100%',
        justifyContent: 'center',
        display: 'flex',
        '&::before, &::after ': {
          alignItems: 'center',
          display: 'flex',
          flex: '1 1',
          content: '""',
          alignSelf: 'center',
          boxSizing: 'border-box',
          height: '1px',
          margin: 'auto',
          backgroundColor: theme.palette.common.black,
        },
      },
      button: {
        alignItems: 'center',
        margin: theme.spacing(15, 4.5),
        backgroundColor: theme.palette.common.black,
        height: '44px',
        borderRadius: '300px',
        '& .MuiButton-label': {
          color: theme.palette.common.white,
          fontWeight: 500,
          fontSize: '16px',
        },
        '& .MuiCircularProgress-circle': {
          color: '#73DDFF',
        },
        '& .MuiCircularProgress-root': {
          marginRight: theme.spacing(2),
        },
        '&:active': {
          boxShadow: 'none',
          backgroundColor: theme.palette.common.black,
        },
        '&:hover': {
          boxShadow: 'none',
          backgroundColor: theme.palette.common.black,
        },
      },
    }),
  { name: 'LoadMoreButton' },
);
