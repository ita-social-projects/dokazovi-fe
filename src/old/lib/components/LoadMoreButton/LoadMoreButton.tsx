import React from 'react';
import { Button, Box, CircularProgress } from '@material-ui/core';
import { LoadingStatusEnum, LoadMoreButtonTextType } from '../../types';
import { LOAD_POSTS_LIMIT } from '../../constants/posts';
import { LOAD_EXPERTS_LIMIT } from '../../constants/experts';
import { useStyles } from './LoadMoreButton.styles';
import { localization } from '../../../../localization';

export interface ILoadMoreButtonProps {
  clicked: () => void;
  textType: LoadMoreButtonTextType;
  loading: LoadingStatusEnum;
  isLastPage: boolean;
  totalPages: number;
  totalElements: number;
  pageNumber: number;
}

function getButtonText(
  count: number,
  textType: LoadMoreButtonTextType,
): string {
  if (textType === LoadMoreButtonTextType.EXPERT) {
    if (count <= 4) {
      return localization.ua.experts.lessThanFour;
    }
    return localization.ua.experts.moreThanFour;
  }
  if (textType === LoadMoreButtonTextType.POST) {
    if (count === 1) {
      return localization.ua.materials.one;
    }
    if (count > 1 && count <= 4) {
      return localization.ua.materials.lessThanFour;
    }
    return localization.ua.materials.moreThanFour;
  }
  return '';
}

function getLimit(textType: LoadMoreButtonTextType): number {
  switch (textType) {
    case LoadMoreButtonTextType.EXPERT:
      return LOAD_EXPERTS_LIMIT;
    case LoadMoreButtonTextType.POST:
      return LOAD_POSTS_LIMIT;
    default:
      return LOAD_POSTS_LIMIT;
  }
}

export const LoadMoreButton: React.FC<ILoadMoreButtonProps> = ({
  clicked,
  loading,
  textType,
  isLastPage,
  pageNumber,
  totalPages,
  totalElements,
}) => {
  const classes = useStyles();
  const renderLoadControls = (): JSX.Element => {
    let control: JSX.Element = <></>;
    const limit = getLimit(textType);
    const isPreLastPage = pageNumber !== totalPages - 2;
    const remainder = totalElements % limit;
    const elementsInLastPage = remainder === 0 ? limit : remainder;
    const elementsInNextPage = isPreLastPage ? limit : elementsInLastPage;
    const buttonText = getButtonText(elementsInNextPage, textType);
    control = (
      <Box className={classes.root}>
        <Button
          className={classes.button}
          variant="outlined"
          onClick={() => {
            clicked();
          }}
        >
          <CircularProgress
            variant={
              loading === LoadingStatusEnum.pending
                ? 'indeterminate'
                : 'determinate'
            }
            color="primary"
            value={80}
            size={25}
            thickness={5}
          />
          Показати ще {elementsInNextPage} {buttonText}
        </Button>
      </Box>
    );

    if (isLastPage) {
      control = <></>;
    }

    return control;
  };
  return renderLoadControls();
};
