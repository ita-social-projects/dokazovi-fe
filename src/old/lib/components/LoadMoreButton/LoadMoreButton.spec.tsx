import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { LoadMoreButton } from './LoadMoreButton';
import { LoadingStatusEnum, LoadMoreButtonTextType } from '../../types';

describe('loadMoreButton test', () => {
  it('rendering when it not last page', () => {
    render(
      <LoadMoreButton
        clicked={jest.fn()}
        textType={LoadMoreButtonTextType.EXPERT}
        loading={LoadingStatusEnum.succeeded}
        isLastPage={false}
        totalPages={2}
        totalElements={24}
        pageNumber={1}
      />,
    );
    expect(screen.getByText('Показати ще 12 авторів')).toBeInTheDocument();
  });
  it('rendering when it is last page', () => {
    const { asFragment } = render(
      <LoadMoreButton
        clicked={jest.fn()}
        textType={LoadMoreButtonTextType.POST}
        loading={LoadingStatusEnum.succeeded}
        isLastPage
        totalPages={1}
        totalElements={12}
        pageNumber={1}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('', () => {
    const loadMoreHandler = jest.fn();
    render(
      <LoadMoreButton
        clicked={loadMoreHandler}
        textType={LoadMoreButtonTextType.POST_BY_STATUS}
        loading={LoadingStatusEnum.succeeded}
        isLastPage={false}
        totalPages={2}
        totalElements={24}
        pageNumber={1}
      />,
    );
    const loadMore = screen.getByTestId('loadMore');
    userEvent.click(loadMore);
    expect(loadMoreHandler).toHaveBeenCalled();
  });
});
