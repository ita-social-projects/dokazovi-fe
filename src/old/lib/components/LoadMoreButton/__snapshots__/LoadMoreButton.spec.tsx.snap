// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loadMoreButton test rendering when it is last page 1`] = `<DocumentFragment />`;
